### PR TITLE
Feature/epmdj 12

### PR DESCRIPTION
**Pull request by s0larm00n**
US3.3. The ability of a user to add new plugins from library to the agent

### CODE REVIEW

File: drill-admin/src/test/kotlin/com/epam/drill/endpoints/ExposeTest.kt [hunk 1]
```diff
@@ -31,18 +35,51 @@ class ExposeTest {
...
-        transaction(connect) {
+        val transaction = transaction(connect) {
             val map = JsonMessages.select {
                 JsonMessages.id.eq("xx")
             }.map { it[JsonMessages.message] }.first()
+
+
             println(map)
         }
+//        transaction{
```
Comment (viktorbll) [inline drill-admin/src/test/kotlin/com/epam/drill/endpoints/ExposeTest.kt]: There should be no junk comments in the code.
Please remove it.

File: drill-admin/src/main/kotlin/com/epam/drill/endpoints/AgentManager.kt [hunk 2]
```diff
@@ -92,6 +83,41 @@ class AgentManager(override val kodein: Kodein) : KodeinAware {
 
     fun byId(agentId: String) = agentStorage.targetMap[agentId]?.first
 
+    suspend fun addPluginFromLib(agentId: String, pluginId: String) {
+        val agentInfoDb = asyncTransaction { AgentInfoDb.findById(agentId) }
+        var agentInfo: AgentInfo? = null
+        if (!(agentInfoDb == null)) {
```
Comment (viktorbll) [inline drill-admin/src/main/kotlin/com/epam/drill/endpoints/AgentManager.kt]: !=

File: drill-admin/src/main/kotlin/com/epam/drill/endpoints/AgentManager.kt [hunk 3]
```diff
@@ -98,10 +93,77 @@ class AgentManager(override val kodein: Kodein) : KodeinAware {
...
+    operator fun contains(k: String) = k in agentStorage.targetMap
+
     fun self(k: String) = agentStorage.self(k)
 
     fun byId(agentId: String) = agentStorage.targetMap[agentId]?.first
 
+    suspend fun addPluginFromLib(agentId: String, pluginId: String) = asyncTransaction {
+        val agentInfoDb = AgentInfoDb.findById(agentId)
+        if (agentInfoDb != null) {
+            val plugin = agentPlugins.getBean(pluginId)!!
```
Comment (IgorKey) [inline drill-admin/src/main/kotlin/com/epam/drill/endpoints/AgentManager.kt]: potential npe

File: drill-agent/src/nativeAgentMain/kotlin/com/epam/drill/core/Starter.kt [hunk 4]
```diff
@@ -89,7 +89,7 @@ private fun runAgent(options: String?) {
 fun String?.asAgentParams(): Map<String, String> {
     if (this.isNullOrEmpty()) return mutableMapOf()
     return try {
-        this.split(",").associate {
+        this!!.split(",").associate {
```
Comment (IgorKey) [inline drill-agent/src/nativeAgentMain/kotlin/com/epam/drill/core/Starter.kt]: wtf?

File: drill-plugin-api/drill-admin-part/src/drillAdminPartMain/kotlin/com/epam/drill/common/AgentInfo.kt [hunk 5]
```diff
@@ -13,8 +13,8 @@ data class AgentInfo(
 
     val adminUrl: String = "",
     var ipAddress: String = "",
-    val rawPluginNames: MutableSet<PluginBean> = mutableSetOf(),
-    val buildVersions: MutableSet<AgentBuildVersionJson>
+    val buildVersions: MutableSet<AgentBuildVersionJson> = mutableSetOf(),
```
Comment (IgorKey) [inline drill-plugin-api/drill-admin-part/src/drillAdminPartMain/kotlin/com/epam/drill/common/AgentInfo.kt]: nice changes...

**Pull request merged**